### ISSUE TITLE
[ISSUE] databricks bundle validate when creating a bundle manually

### ISSUE THREAD

**Issue by christopher-danz** [databricks/databricks-sdk-go#796]
**Description**
Calling databricks bundles validate returns the following error:

Error: unexpected error handling request: json: cannot unmarshal number into Go struct field APIErrorBody.error_code of type string. This is likely a bug in the Databricks SDK for Go or the underlying REST API. Please report this issue with the following debugging information to the SDK issue tracker at https://github.com/databricks/databricks-sdk-go/issues. 

**Reproduction**
Going through these guides ... 
https://docs.databricks.com/en/dev-tools/bundles/work-tasks.html
https://docs.databricks.com/en/dev-tools/bundles/settings.html#examples 
... to create a Databricks Asset Bundle manually. 

1. Create empty repository in VSCode
2. Install YAML language server support extension
3. Generate Databricks Asset Bundle configuration JSON schema file by running:
`databricks bundle schema > bundle_config_schema.json`
4. Use VSCode to create or open a bundle configuration file within the current directory, name it "databricks.yml"
5. Add the following comment to the beginning of your bundle configuration file:
`# yaml-language-server: $schema=bundle_config_schema.json`
6. Fill in the bundle configuration files, e.g by using the contents from this source: https://docs.databricks.com/en/dev-tools/bundles/settings.html#examples
like this content: 
```
bundle:
  name: hello-bundle

targets:
  dev:
    default: true
    resources:
      jobs:
        hello-job:
          name: hello-job
          tasks:
            - task_key: hello-task
              existing_cluster_id: 1234-567890-abcde123
              notebook_task:
                notebook_path: ./hello.py
```
7. Validate the bundle configuration files by running:
`databricks bundle validate`

**Expected behavior**
Some kind of output string that indicates the configuration file is synthetically correct. (Can't further specify since it is nowhere written how the correct execution of the command looks like or what it outputs)

**Is it a regression?**
Using Databricks CLI v0.212.2

**Debug Logs**
Request log:
```
GET /api/2.0/preview/scim/v2/Me
> * Host:
> * Accept: application/json
> * Authorization: REDACTED
> * User-Agent: cli/0.212.2 databricks-sdk-go/0.29.1 go/1.21.6 os/windows cmd/bundle_validate auth/pat
< HTTP/2.0 403 Forbidden
< * Content-Length: 52
< * Content-Type: application/json; charset=utf-8
< * Date: Thu, 01 Feb 2024 10:37:20 GMT
< * Server: databricks
< * Vary: Accept-Encoding
< * X-Databricks-Reason-Phrase: Invalid access token.
< {
<   "error_code": 403,
<   "message": "Invalid access token."
< }
```

**Other Information**
 - OS: Windows
 - Version: 10

**Additional context**

**Comment by christopher-danz** [comment]
I tried everything there is to troubleshoot.. empty cache, create new .databrickscfg file, generate new token, ... the whole thing and I am not entirely sure what made the differnce in the end..

After ensuring you authenticate correctly, try to run:
`databricks jobs list`
... it should list all the jobs for your DEFAULT profile.

then run:
`databricks jobs list --profile=YOUR_DESIRED_PROFILE_TO_USE_DAB`
and it should list the jobs for the other host.

I then ran:
`databricks bundle init` (specified the unique name and the additional stuff)
and it worked then.

**Comment by christopher-danz** [comment]
I tried everything there is to troubleshoot.. empty cache, create new .databrickscfg file, generate new token, ... the whole thing and I am not entirely sure what made the differnce in the end..

After ensuring you authenticate correctly, try to run:
`databricks jobs list`
... it should list all the jobs for your DEFAULT profile.

then run:
`databricks jobs list --profile=YOUR_DESIRED_PROFILE_TO_USE_DAB`
and it should list the jobs for the other host.

I then ran:
`databricks bundle init `(specified the unique name and the additional stuff)
and it worked then.
